cordova-lib/src/configparser/ConfigParser.js

```diff
@@ -219,6 +258,47 @@ ConfigParser.prototype = {
     },
 
     /**
+     * Finds the value of the orientation <preference>.
+     *
+     * @param  {String} [platform]  Platform name
+     * @return {String}             Global/platform-specific orientation (or undefined if both are undefined)
+     */
+    getOrientation: function (platform) {
+
+        var globalOrientation = this.getGlobalPreference('orientation');
+        var platformOrientation = null;
+        var platformSupportedOrientations = null;
+
+        // Check if the value of the global orientation preference is invalid
+        if (globalOrientation && !_.contains(PREFERENCE_ORIENTATION_VALUES, globalOrientation)) {
+            events.emit( 'warn', [ 'Unknown value for global orientation preference:', globalOrientation ].join(' ') );
+            globalOrientation = undefined;
+        }
+
+        if (!platform) {
+            return globalOrientation;
+        }
+
+        platformSupportedOrientations = supportedPreferences.orientation[platform];
+
+        if (!platformSupportedOrientations) {
+            throw new CordovaError([ 'No supported orientations defined for platform', platform ].join (' '));
+        }
+
+        platformOrientation = this.getPlatformPreference('orientation', platform);
+
+        // Check if the value of the platform-specific orientation preference is invalid
+        if (platformOrientation && !_.contains(platformSupportedOrientations, platformOrientation)) {
```

Comment:
Yep, that would've been a better approach. In the end supportedOrientations.json seemed to be useful only for validating user input so I omitted this check for platform-specific support altogether.